### PR TITLE
NXP-26690: transactional DBS

### PR DESCRIPTION
**Pull request by efge**

### CODE REVIEW

File: nuxeo-core/nuxeo-core-storage-dbs/src/main/java/org/nuxeo/ecm/core/storage/dbs/DBSTransactionState.java [hunk 1]
```diff
@@ -151,13 +161,24 @@
...
+        this.connection = connection;
         this.session = session;
         SecurityService securityService = Framework.getService(SecurityService.class);
         browsePermissions = new HashSet<>(Arrays.asList(securityService.getPermissionsToCheck(BROWSE)));
     }
 
+    /** @since 11.1 */
+    public void close() {
+        connection.close();
+    }
```
Comment (kevinleturc) [inline nuxeo-core/nuxeo-core-storage-dbs/src/main/java/org/nuxeo/ecm/core/storage/dbs/DBSTransactionState.java]: Can we make DBSTransactionState a `Closeable`?

File: nuxeo-core/nuxeo-core-storage-dbs/src/main/java/org/nuxeo/ecm/core/storage/dbs/DBSSession.java [hunk 2]
```diff
@@ -192,7 +192,9 @@
 
     public DBSSession(DBSRepository repository) {
         this.repository = repository;
-        transaction = new DBSTransactionState(repository, this);
+        @SuppressWarnings("resource") // connection closed by DBSTransactionState.close
+        DBSConnection connection = repository.getConnection();
+        transaction = new DBSTransactionState(repository, connection, this);
```
Comment (kevinleturc) [inline nuxeo-core/nuxeo-core-storage-dbs/src/main/java/org/nuxeo/ecm/core/storage/dbs/DBSSession.java]: I didn't find any other instantiation of DBSTransactionState which makes me think we don't need to change the DBSTransactionState constructor and just do the getConnection into it, wdy?

Comment (efge) [reply nuxeo-core/nuxeo-core-storage-dbs/src/main/java/org/nuxeo/ecm/core/storage/dbs/DBSSession.java]: Yes I'll do that

File: nuxeo-core/nuxeo-core-storage-dbs/src/main/java/org/nuxeo/ecm/core/storage/dbs/DBSTransactionState.java (hunk 1 above)
Comment (efge) [reply nuxeo-core/nuxeo-core-storage-dbs/src/main/java/org/nuxeo/ecm/core/storage/dbs/DBSTransactionState.java]: Yes I don't think it's a problem.

Comment (efge) [reply nuxeo-core/nuxeo-core-storage-dbs/src/main/java/org/nuxeo/ecm/core/storage/dbs/DBSTransactionState.java]: `Autocloseable` actually has this really has nothing to do with `java.io`

**Comment by efge** [comment]
Updated after @kevinleturc's comments

**Pull request merged**